backbone.js

```diff
@@ -1140,6 +1140,7 @@
         removed.push(model);
         this._removeReference(model, options);
       }
+      if (models.length > 0 && !options.silent) delete options.index;
```

Comment:
Could also add `index:void 0` as part of the `setOptions` which would eliminate case 3 and then a `options.index = void 0;` in `_removeModels` would result in consistent behavior.